Lib/site.py

```diff
@@ -168,6 +169,14 @@ def addpackage(sitedir, name, known_paths):
     else:
         reset = False
     fullname = os.path.join(sitedir, name)
+    try:
+        st = os.lstat(fullname)
+    except OSError:
+        return
```

Comment:
I'm slightly worried about the performance impact of calling stat on all `.pth` files.  `pip install -e` will install a `.pth` file for every installed editable package.

On the other hand, the overhead should be neglectable as the inode needs to be read in memory anyway to actually read the contents.